src/components/withEnvironment.js

```diff
@@ -0,0 +1,44 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import Environment from '../libs/Environment';
+import CONST from '../CONST';
+
+const environmentPropTypes = {
+    /** The string value representing the current environment */
+    environment: PropTypes.string.isRequired,
+};
+
+export default function (WrappedComponent) {
+    class WithEnvironment extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                environment: CONST.ENVIRONMENT.PRODUCTION,
+            };
+        }
+
+        componentDidMount() {
+            Environment.getEnvironment()
+                .then((environment) => {
+                    this.setState({environment});
+                });
+        }
+
+        render() {
+            return (
+                <WrappedComponent
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...this.props}
+                    environment={this.state.environment}
+                />
+            );
+        }
+    }
+
+    return WithEnvironment;
```

Comment:
I've left out the `displayName` value since this is a class component, but just to double-check, does the fact that this is a Higher-Order Component change anything? For example, I see that [WithDrawerState](https://github.com/Expensify/Expensify.cash/blob/7fa89be020e070867ce9426fb02697bba8e21a3b/src/components/withDrawerState.js#L23) and [WithLocalize](https://github.com/Expensify/Expensify.cash/blob/37465dbd07da1feab8347835d82ed3d2302cde4c/src/components/withLocalize.js#L60) both set a displayName, but that makes sense since they are functional. [WithWindowDimensions](https://github.com/Expensify/Expensify.cash/blob/3967032d964325f17c3cb0326211567f7a731d19/src/components/withWindowDimensions.js#L91) on the other hand looks like a class component, but still has the definition, so in this case should I remove the displayName from withWindowDimensions? Or do I need to add displayName to withEnvironment?

Reply:
Yeah, so the whole point of `displayName` is to get components to be understandable in the React Dev Tools. When it comes to the HOCs, every HOC, regardless of it being functional or a class component, it should still redefine the `displayName` to say `with<NameOfHOC>(<NameOfWrappedComponent>)`. That way it is clear from looking at the dev tools which components are the HOCs and which ones aren't.

That's why you see all of those HOCs setting the display name and why you should do the same thing here.